lib/src/entrypoint.dart

```diff
@@ -62,21 +63,78 @@ class Entrypoint {
   // true.
   final String workingDir;
 
-  Package? _workspaceRoot;
+  static ({Package root, Package work}) _loadWorkspace(
+    String workingDir,
+    SystemCache cache,
+  ) {
+    if (!dirExists(workingDir)) {
+      fail('The directory `$workingDir` does not exist.');
+    }
+    // Keep track of all the pubspecs met when walking up the file system.
+    // The first of these is the workingPackage.
+    final pubspecsMet = <String, Pubspec>{};
+    for (final dir in parentDirs(workingDir)) {
+      final Pubspec pubspec;
+
+      try {
+        pubspec = Pubspec.load(
+          dir,
+          cache.sources,
+          containingDescription: RootDescription(dir),
+          allowOverridesFile: true,
+        );
+      } on FileException {
+        continue;
+      }
+      pubspecsMet[p.canonicalize(dir)] = pubspec;
+      final Package root;
+      if (pubspec.resolution == Resolution.none) {
+        root = Package.load(
+          null,
+          dir,
+          cache.sources,
+          alreadyLoadedPubspecs: pubspecsMet,
+          withPubspecOverrides: true,
```

Comment:
Are you reloading it because it's the root, and it's allowed to a have `pubspec_overrides.yaml` ?

If so, can we maybe highlight that in a comment.

Reply:
I'm not reloading. This is where the root is loaded (actually created).

This is the `Package1 object. What we have already loaded is the `Pubspec` object.
The `Pubspec` will be in the `pubspecsMet` map, so we will not actually hit the filesystem again for the root pubspec (but we might for other pubspecs in the workspace).